### PR TITLE
Make PointBuilder, CircleBuilder & EnvelopeBuilder implement Writable 

### PR DESCRIPTION
**Pull request by cbuescher**
After refactoring the queries to make them parsable on the coordinating note and adding serialization and equals/hashCode capability to them. So far ShapeBuilders nested inside queries were still transported as a byte array that needs to be parsed later on the shard receiving the query. To be able to also serialize geo shapes this way, we also need to make all the implementations of ShapeBuilder implement Writable. This PR adds this to PointBuilder, CircleBuilder and EnvelopeBuilder and also adds tests for serialization, equality and hashCode.

Relates to #14416

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/common/geo/builders/CircleBuilder.java [hunk 1]
```diff
@@ -116,4 +143,35 @@ public Circle build() {
...
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        writeCoordinateTo(center, out);
+        out.writeDouble(radius);
+        DistanceUnit.writeDistanceUnit(out, unit);
+    }
+
+    @Override
+    public ShapeBuilder readFrom(StreamInput in) throws IOException {
+        return new CircleBuilder().center(readCoordinateFrom(in)).radius(in.readDouble(), DistanceUnit.readDistanceUnit(in));
```
Comment (nknize) [inline core/src/main/java/org/elasticsearch/common/geo/builders/CircleBuilder.java]: nit: can you break this into multiple lines so its easier to track w/ ```writeTo```

File: core/src/test/java/org/elasticsearch/common/geo/builders/PointBuilderTests.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+package org.elasticsearch.common.geo.builders;
+
+import com.vividsolutions.jts.geom.Coordinate;
+
+import org.elasticsearch.test.geo.RandomShapeGenerator;
+import org.elasticsearch.test.geo.RandomShapeGenerator.ShapeType;
+
+public class PointBuilderTests extends AbstractShapeBuilderTestCase<PointBuilder> {
```
Comment (nknize) [inline core/src/test/java/org/elasticsearch/common/geo/builders/PointBuilderTests.java]: MultiPoint tests?

Comment (nknize) [inline core/src/test/java/org/elasticsearch/common/geo/builders/PointBuilderTests.java]: ah, I suppose those will be in a separate tests for now. Maybe we can revisit later how to cleanly combine MultiPoint and Point into one builder. Same with Multi {LineString | Polygon}

**Comment by nknize** [comment]
This looks good to me. In serialization logic I'd like to see each readFrom on their own line (just like writeTo). Maybe its just me but its easier to read/maintain when you can track one-to-one read/write.

File: core/src/main/java/org/elasticsearch/common/geo/builders/CircleBuilder.java (hunk 1 above)
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/common/geo/builders/CircleBuilder.java]: sure, will do.

File: core/src/test/java/org/elasticsearch/common/geo/builders/PointBuilderTests.java (hunk 2 above)
Comment (cbuescher) [inline core/src/test/java/org/elasticsearch/common/geo/builders/PointBuilderTests.java]: Yes, I have those tests almost ready to got with MultiPointBuilder. Happy to talk about possibilities to combine those two in follow-up PRs.

**Comment by cbuescher** [comment]
@nknize thanks for the review, adressed your comments and will merge.

**Pull request merged**